### ISSUE TITLE
Generating .ovpn with the same ip

### ISSUE THREAD

**Issue by RaddadZ** [simonwep/openvpn-pihole#1]
Hi, first of all, I really appreciate the work you have done here. 
I'm facing a problem with .ovpn files, as every file I generate despite having a different unique name it always connects with the same IP.

I did update the host in **client configuration**. I only changed the remote IP before using this command:
`docker exec openvpn bash /opt/app/bin/genclient.sh <name> <password?>`

this is what I get:

```
root@docker-ubuntu:~# sudo docker exec openvpn bash /opt/app/bin/genclient.sh raddadz-test
Generate client certificate...
Without password...
Using SSL: openssl OpenSSL 1.1.1  11 Sep 2018
Generating a RSA private key
........................................................................+++++
......+++++
writing new private key to '/opt/app/easy-rsa/pki/easy-rsa-26.Ie6cbd/tmp.KImTmr'
-----
./easyrsa: 341: set: Illegal option -o echo
Using SSL: openssl OpenSSL 1.1.1  11 Sep 2018
Using configuration from /opt/app/easy-rsa/pki/easy-rsa-49.6jqTTg/tmp.1yoe0I
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'ChangeMe'
Certificate is to be certified until Dec 22 06:21:36 2022 GMT (825 days)

Write out database with 1 new entries
Data Base Updated
./easyrsa: 341: set: Illegal option -o echo
/opt/app/bin/genclient.sh: line 42: warning: command substitution: ignored null byte in input
Sync pki directory...
Generate .ovpn file...
OpenVPN Client configuration successfully generated!
Checkout <service-name>/clients/raddadz-test.ovpn
```

there are two lines I think causing the problem:
`./easyrsa: 341: set: Illegal option -o echo`
and 
`commonName            :ASN.1 12:'ChangeMe'`

I'm still new to OpenVPN and certification generation, but I looked into this problem and my results as follows:

- I've found the same IP problem related to using the same client certificate to generate files, and the suggested solution was using ccd files for each client cert. I don't think this the problem though, since I think this repo uses different client certificates.
- there are old issues opened may be related to this issue, [here](https://github.com/OpenVPN/easy-rsa/issues/73) which suggests changing `#set_var EASYRSA_REQ_CN	"ChangeMe"` in vars. 
- there is also another issue talking about ` set: Illegal option -o echo` [here](https://github.com/OpenVPN/easy-rsa/issues/308)
- also changing `;duplicate-cn` in server.conf might solve the problem, but I don't think it's the problem, because the generated .ovpn files have different `<cert>` and `<key>` values

I'm not sure what to do as I've tried a couple of solutions and I wasn't able to get different IPs for different clients.
Your guidance will be appreciated.

#### my environment (if relevant)
 - OS: Ubuntu 20.04.1 LTS
 - Docker version 19.03.12
 - docker-compose version 1.22.0
 - Pi-hole v5.1.2
 - OpenVPN 2.4.4

**Comment by simonwep** [comment]
First of all, thank you for this elaborated issue!

The cli output you've posted looks correct to me (although I should probably fix `/opt/app/bin/genclient.sh: line 42: warning: command substitution: ignored null byte in input`).

Yes `commonName :ASN.1 12:'ChangeMe'` is the problem - since I'm using the same `.ovpn` file for all my devices (in combination with `duplicate-cn`) I didn't notice that. I will see if I can use [this method](https://github.com/OpenVPN/easy-rsa/issues/73#issuecomment-223170538) to use the client-name as commonName too. This is definitely a bug and something I forgot to add, I'll take a look into it :)

**Comment by simonwep** [comment]
I fixed the bash warning in 3213fa3e23183350eddb55315c4816fbcf9c5195 and the common-name issue in 74a8cb6ef9bfaacca44194787ea2e6caee81c86f, if it works on your end too we can close this issue :)

**Comment by RaddadZ** [comment]
it works like a charm! `duplicate-cn` worked for me but I prefare to use different .ovpn files for my different devices, so this is what I needed, thanks.
